pkg/jobs/adopt.go

```diff
@@ -426,11 +428,28 @@ func (r *Registry) runJob(
 	// NB: After this point, the job may no longer have the claim
 	// and further updates to the job record from this node may
 	// fail.
-	r.maybeClearLease(job, err)
-	r.maybeDumpTrace(ctx, resumer, job.ID())
+	grp := ctxgroup.WithContext(ctx)
+	if err != nil {
+		grp.GoCtx(func(ctx context.Context) error {
+			// We delay clearing the lease to avoid situations where a job fast-fails
+			// and is immediately re-adopted by another node, causing a hot loop of job
+			// status changes. See #158597 for more info.
```

Comment:
I think we might also reminder the reader that _our_ node that holds the claim won't re-run this job while we are waiting for this interval because the job is not removed from the list of currently running jobs until the enclosing function returns.